speaker/main.go

```diff
@@ -368,17 +376,29 @@ func (c *controller) SetNode(l gokitlog.Logger, node *v1.Node) k8s.SyncState {
 	return k8s.SyncStateSuccess
 }
 
+func (c *controller) SetSpeakers(l gokitlog.Logger, eps k8s.EpsOrSlices) k8s.SyncState {
+	for proto, handler := range c.protocols {
+		if err := handler.SetSpeakers(l, eps); err != nil {
+			l.Log("op", "SetSpeakers", "error", err, "protocol", proto, "msg", "failed to propagate speakers info to protocol handler")
+			return k8s.SyncStateError
+		}
+	}
+	return k8s.SyncStateSuccess
+}
+
 // A Protocol can advertise an IP address.
 type Protocol interface {
 	SetConfig(gokitlog.Logger, *config.Config) error
 	ShouldAnnounce(gokitlog.Logger, string, *v1.Service, k8s.EpsOrSlices) string
 	SetBalancer(gokitlog.Logger, string, net.IP, *config.Pool) error
 	DeleteBalancer(gokitlog.Logger, string, string) error
 	SetNode(gokitlog.Logger, *v1.Node) error
+	SetSpeakers(gokitlog.Logger, k8s.EpsOrSlices) error
```

Comment:
in a future change (not this PR), it would be nice to write docstrings for all of these interface methods. The interfaces have grown a lot since the original design, and it's no longer obvious to me what these things are expected to do (probably my fault for not documenting originally, sorry)